controller/uploadController.js

```diff
@@ -0,0 +1,72 @@
+const {
+  uploadCloudinary,
+  rollbackUploadCloudinary,
+} = require("../lib/cloudinary");
+const fs = require("fs");
+
+class Upload {
+  static async uploadFile(req, res) {
+    let statusCode = 201;
+    let result = {
+      message: "Success",
+      data: null,
+    };
+
+    try {
+      let uploadResult = await uploadCloudinary(req.file.path);
+      result.data = uploadResult;
+
+      fs.unlink(req.file.path, (err) => {
+        if (err) {
+          result.message = err.message;
+          statusCode = 500;
+        }
+      });
+    } catch (error) {
+      result.message = error.message;
+      statusCode = 500;
+      rollbackUploadCloudinary(uploadResult.public_id);
```

Comment:
no need to rollback the upload proocess, and uploadResult variable will be undefined